primitives/src/vid/advz.rs

```diff
@@ -218,124 +217,15 @@ where
-            &V::Index::from(share.index as u64),
-            &share.evals_proof,
-        )
-        .map_err(vid)?
-        .is_err()
-        {
-            return Ok(Err(()));
-        }
-
-        let pseudorandom_scalar = Self::pseudorandom_scalar(common)?;
-
-        // Compute aggregate polynomial [commitment|evaluation]
-        // as a pseudorandom linear combo of [commitments|evaluations]
-        // via evaluation of the polynomial whose coefficients are
-        // [commitments|evaluations] and whose input point is the pseudorandom
-        // scalar.
-        let aggregate_poly_commit = P::Commitment::from(
-            polynomial_eval(
-                common
-                    .poly_commits
-                    .iter()
-                    .map(|x| CurveMultiplier(x.as_ref())),
-                pseudorandom_scalar,
-            )
-            .into(),
-        );
-        let aggregate_eval =
-            polynomial_eval(share.evals.iter().map(FieldMultiplier), pseudorandom_scalar);
-
-        // verify aggregate proof
-        Ok(P::verify(
-            &self.vk,
-            &aggregate_poly_commit,
-            &self.multi_open_domain.element(share.index),
-            &aggregate_eval,
-            &share.aggregate_proof,
-        )
-        .map_err(vid)?
-        .then_some(())
-        .ok_or(()))
-    }
-
-    fn recover_payload(&self, shares: &[Self::Share], common: &Self::Common) -> VidResult<Vec<u8>> {
-        // TODO can we avoid collect() here?
-        Ok(field_to_bytes(self.recover_elems(shares, common)?).collect())
-    }
-}
+    fn disperse(&self, payload: &Self::Payload) -> VidResult<VidDisperse<Self>> {
+        let bytes_len = payload.as_slice().len();
 
-impl<P, T, H, V> GenericAdvz<P, T, H, V>
-where
-    P: UnivariatePCS<Point = <P as PolynomialCommitmentScheme>::Evaluation>,
-    P::Evaluation: PrimeField,
-    P::Polynomial: DenseUVPolynomial<P::Evaluation>,
-    P::Commitment: From<T> + AsRef<T>,
-    T: AffineRepr<ScalarField = P::Evaluation>,
-    H: Digest + DynDigest + Default + Clone + Write,
-    V: MerkleTreeScheme<Element = Vec<P::Evaluation>>,
-    V::MembershipProof: Sync + Debug, /* TODO https://github.com/EspressoSystems/jellyfish/issues/253 */
-    V::Index: From<u64>,
-{
-    /// Same as [`VidScheme::disperse`] except `payload` iterates over
-    /// field elements.
-    pub fn disperse_from_elems<I>(&self, payload: I) -> VidResult<VidDisperse<Self>>
-    where
-        I: IntoIterator,
-        I::Item: Borrow<P::Evaluation>,
-    {
         // partition payload into polynomial coefficients
         // and count `elems_len` for later
-        let elems_iter = payload.into_iter().map(|elem| *elem.borrow());
-        let mut elems_len = 0;
+        let elems_iter = bytes_to_field::<_, P::Evaluation>(payload).map(|elem| *elem.borrow());
         let mut polys = Vec::new();
         for coeffs_iter in elems_iter.chunks(self.payload_chunk_size).into_iter() {
```

Comment:
Same here, maybe better to change the variable name of `coeffs_iter` to `evals_iter`.